### PR TITLE
Implement nighly build test mechanism

### PR DESCRIPTION
**Pull request by hukoyu**
With this PR we can mark any test (long running ones for example) as nightly build only
by adding them to file .azure-pipelines/other/nighlty_run_only_tests.txt

If SGXLKL_NIGHTLY_BUILD environment variable is set to false, the tests in this file will be skipped.
These tests will run only if SGXLKL_NIGHTLY_BUILD env variable is true.

### CODE REVIEW

File: .azure-pipelines/scripts/test_runner.sh [hunk 1]
```diff
@@ -18,6 +18,10 @@ if [ -z "$SGXLKL_BUILD_MODE" ]; then
     echo "ERROR: 'SGXLKL_BUILD_MODE' is undefined. Please export SGXLKL_BUILD_MODE=<mode>"
     exit 1
 fi
+if [ -z "$SGXLKL_NIGHTLY_BUILD" ]; then
```
Comment (jxyang) [inline .azure-pipelines/scripts/test_runner.sh:21]: If not defined, we should treat as non-nightly build.

File: .azure-pipelines/scripts/test_runner.sh [hunk 2]
```diff
@@ -134,12 +138,24 @@ function GetReadyToRunNextTest()
...
+    if [[ $is_test_disabled -eq 0 && $SGXLKL_NIGHTLY_BUILD -eq 0 ]]; then
+        is_test_nightly_only=$(grep -c "$file" "$nightly_tests_file")
+        if [[ $is_test_nightly_only -ge 1 ]]; then
+            echo "Test $file is marked nightly build only. Skipping test..."
+            echo "To enable the test remove $file from $nightly_tests_file"
+        fi
+    fi
+
+    # if this test is disabled set counters and skip to next test
+    if [[ $is_test_disabled -ge 1 || $is_test_nightly_only -ge 1 ]]; then
```
Comment (jxyang) [inline .azure-pipelines/scripts/test_runner.sh]: `is_test_nightly_only -ge 1` is confusing. You can remove it by setting `is_test_disabled ` on line 152.

File: .azure-pipelines/scripts/test_runner.sh [hunk 3]
```diff
@@ -183,7 +200,7 @@ suite_test_start_time=$(date +%s)
 
 failure_identifiers=()
 while IFS= read -r line; do failure_identifiers+=("$line"); done < "$failure_identifiers_file"
-
+echo "This is Nightly Build: $SGXLKL_NIGHTLY_BUILD"
```
Comment (jxyang) [inline .azure-pipelines/scripts/test_runner.sh]: Confusing message. Suggest only printing when $SGXLKL_NIGHTLY_BUILD is set to 1.

File: .azure-pipelines/scripts/junit_utils.sh [hunk 4]
```diff
@@ -136,12 +136,12 @@ function JunitTestFinished()
         fi
 	# Replace STACK_TRACE place holder with stack trace file
         AddLogFileToJunit "$stack_trace_file_path" "$junit_file_path" "STACK_TRACE"
-    fi
 
-    # Add stdout file to junit if exist
-    if [[ -f "$stdout_file_path" ]]; then
-        echo "      <system-out>STD_OUT_MESSAGE</system-out>" >> "$junit_file_path"
-        AddLogFileToJunit "$stdout_file_path" "$junit_file_path" "STD_OUT_MESSAGE"
+        # Add stdout file to junit if exist
```
Comment (jxyang) [inline .azure-pipelines/scripts/junit_utils.sh:140]: This is not related to nightly build, right?

File: .azure-pipelines/scripts/test_runner.sh (hunk 1 above)
Comment (jxyang) [inline .azure-pipelines/scripts/test_runner.sh:21]: Why do we force another environment? Can we treat not defined as 0?

File: .azure-pipelines/scripts/test_runner.sh (hunk 2 above)
Comment (jxyang) [inline .azure-pipelines/scripts/test_runner.sh]: `$is_test_nightly_only -ge 1` is confusing. You can get rid of it by setting `$is_test_disabled` on line 152.

File: .azure-pipelines/scripts/test_runner.sh (hunk 3 above)
Comment (jxyang) [inline .azure-pipelines/scripts/test_runner.sh]: Confusing message. Suggest rewording.

File: .azure-pipelines/scripts/junit_utils.sh (hunk 4 above)
Comment (hukoyu) [reply .azure-pipelines/scripts/junit_utils.sh:140]: Yes this not directly related but there is an issue with downloaded html page encoding with basic/networking test that blocks junit xml file and all core test junit file publish. To be able to see test results I need to fix this. This can be a separate PR but if it is OK I want to fix it as part of this PR.

File: .azure-pipelines/scripts/test_runner.sh (hunk 3 above)
Comment (hukoyu) [reply .azure-pipelines/scripts/test_runner.sh]: Done removed. This message is not necessary actually.

File: .azure-pipelines/scripts/test_runner.sh (hunk 2 above)
Comment (hukoyu) [reply .azure-pipelines/scripts/test_runner.sh]: Done. Got rid of is_test_nightly_only  and used is_test_disabled

File: .azure-pipelines/scripts/test_runner.sh (hunk 1 above)
Comment (hukoyu) [reply .azure-pipelines/scripts/test_runner.sh:21]: Done

File: .azure-pipelines/scripts/test_runner.sh (hunk 3 above)
Comment (hukoyu) [reply .azure-pipelines/scripts/test_runner.sh]: Make sense. Printed message for 1 only. Did rewording.

File: .azure-pipelines/scripts/test_runner.sh (hunk 2 above)
Comment (hukoyu) [reply .azure-pipelines/scripts/test_runner.sh]: Done

File: .azure-pipelines/scripts/test_runner.sh (hunk 1 above)
Comment (hukoyu) [reply .azure-pipelines/scripts/test_runner.sh:21]: Done

**Review by jxyang — APPROVED** [review]

File: .azure-pipelines/template.yml [hunk 5]
```diff
@@ -47,6 +47,9 @@ parameters:
 - name: 'publish'
   type: boolean
   default: false
+- name: 'nightly_build'
+  type: number
```
Comment (letmaik) [inline .azure-pipelines/template.yml:51]: Why not a boolean? This would be cleaner I think. You could then conditionally include the env var:
```yaml
            - bash: .azure-pipelines/scripts/test_runner.sh ${{ test_suite }}
              displayName: Run Tests
              env:
                ..
                {{ if parameters.nightly_build }}:
                  SGXLKL_NIGHTLY_BUILD: 1
```
